clients/src/main/java/org/apache/kafka/common/requests/BrokerRegistrationRequest.java

```diff
@@ -46,21 +46,7 @@ public short oldestAllowedVersion() {
 
         @Override
         public BrokerRegistrationRequest build(short version) {
-            if (version < 4) {
-                // Workaround for KAFKA-17011: for BrokerRegistrationRequest versions older than 4,
-                // translate minSupportedVersion = 0 to minSupportedVersion = 1.
-                BrokerRegistrationRequestData newData = data.duplicate();
-                for (Iterator<BrokerRegistrationRequestData.Feature> iter = newData.features().iterator();
-                     iter.hasNext(); ) {
-                    BrokerRegistrationRequestData.Feature feature = iter.next();
-                    if (feature.minSupportedVersion() == 0) {
-                        feature.setMinSupportedVersion((short) 1);
-                    }
-                }
-                return new BrokerRegistrationRequest(newData, version);
-            } else {
-                return new BrokerRegistrationRequest(data, version);
-            }
+            return new BrokerRegistrationRequest(data, version);
```

Comment:
We need to skip features with minSupportedVersion of 0 in v3 or less.

We need to do the same in ApiVersionResponse.

We need to change the comment like the following in ApiVersionResponse.json and BrokerRegistrationRequest.json.

`Note: in v0-v3, features with MinSupportedVersion = 0 show up with MinSupportedVersion = 1.`

Reply:
for another: please update the following comment:
```
                // Workaround for KAFKA-17011: for BrokerRegistrationRequest versions older than 4,
                // translate minSupportedVersion = 0 to minSupportedVersion = 1.
```